network/ovssnat/ovssnat.go

```diff
@@ -0,0 +1,411 @@
+package ovssnat
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/Azure/azure-container-networking/iptables"
+	"github.com/Azure/azure-container-networking/log"
+	"github.com/Azure/azure-container-networking/netlink"
+	"github.com/Azure/azure-container-networking/network/epcommon"
+
+	"github.com/Azure/azure-container-networking/ovsctl"
+	"github.com/Azure/azure-container-networking/platform"
+)
+
+const (
+	azureSnatVeth0  = "azSnatveth0"
+	azureSnatVeth1  = "azSnatveth1"
+	azureSnatIfName = "eth1"
+	cniOutputChain  = "AZURECNIOUTPUT"
+	cniInputChain   = "AZURECNIINPUT"
+	SnatBridgeName  = "azSnatbr"
+	ImdsIP          = "169.254.169.254/32"
+)
+
+type OVSSnatClient struct {
+	hostSnatVethName       string
+	containerSnatVethName  string
+	localIP                string
+	snatBridgeIP           string
+	SkipAddressesFromBlock []string
+	containerSnatVethMac   net.HardwareAddr
+}
+
+func NewSnatClient(hostIfName string, contIfName string, localIP string, snatBridgeIP string, skipAddressesFromBlock []string) OVSSnatClient {
+	log.Printf("Initialize new snat client")
+	snatClient := OVSSnatClient{}
```

Comment:
you can also do something like
OVSSnatClient{
hostSnatVethName: hotIfName,
...
}